rpm/cloud-hypervisor.spec

```diff
@@ -1,25 +1,32 @@
-# This spec file assumes you're building on an environment where:
-# * You have access to the internet during the build
-# * You have rustup installed on your system
-# * You have both x86_64-unknown-linux-gnu and  x86_64-unknown-linux-musl
-#   targets installed.
-
+# If this flag is set to 1, rustup installation needs to exist on the system and
+# <arch>-unknown-linux-gnu target is required.
+# If this flag is set to 0, distro specific Rust packages will be pulled into the build environment.
 %define using_rustup 1
+# If this flag is set to 1, <arch>-unknown-linux-musl target is required.
 %define using_musl_libc 1
+# If this flag is set to 1, the vendored crates archive and cargo.toml need to be prepared and
+# offline build is implied. Attached script update_src can be used for the vendorization.
+# If this flag is set to 0, access to the internet is required during the build.
+%define using_vendored_crates 0
 
 Name:           cloud-hypervisor
 Summary:        Cloud Hypervisor is an open source Virtual Machine Monitor (VMM) that runs on top of KVM.
-Version:        23.0
+Version:        23.1
 Release:        0%{?dist}
 License:        ASL 2.0 or BSD-3-clause
 Group:          Applications/System
 Source0:        https://github.com/cloud-hypervisor/cloud-hypervisor/archive/v%{version}.tar.gz
-ExclusiveArch:  x86_64
+%if 0%{?using_vendored_crates}
+Source1:        vendor.tar.gz
+Source2:        config.toml
+%endif
+ExclusiveArch:  x86_64 aarch64
 
 BuildRequires:  gcc
 BuildRequires:  glibc-devel
 BuildRequires:  binutils
 BuildRequires:  git
+BuildRequires:  openssl-devel
```

Comment:
Why do we required openssl-devel?  Would be nice to at least menton why it's been added here as part of the commit message.

Reply:
Yeah, the explanation is a couple of lines lower, see `OPENSSL_NO_VENDOR` to affect `openssl-sys` behaviors. The goal here is to utilize the distro OpenSSL instead of rebuilding.

Thansk.